File: src/backend/app/schemas/seat.py
# backend/app/schemas/seat.py
from typing import Optional

from pydantic import BaseModel


class SeatResponse(BaseModel):
    id: str
    label_zh: str
    label_en: str
    zone: str
    status: str
    is_active: bool

    class Config:
        from_attributes = True


class SeatStatusUpdate(BaseModel):
    status: str


class SeatListResponse(BaseModel):
    seats: list[SeatResponse]

File: src/frontend/src/types/seat.ts
// Types for Seat
export interface Seat {
  id: string
  label_zh: string
  label_en: string
  zone: 'indoor' | 'outdoor' | 'bar'
  status: 'vacant' | 'occupied' | 'reserved' | 'inactive'
  is_active: boolean
}

export interface SeatsResponse {
  seats: Seat[]
}
